modules/dnn/src/layers/accum_layer.cpp

```diff
@@ -0,0 +1,168 @@
+        }
+
+        outputs.assign(1, outShape);
+        return false;
+    }
+
+    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        const int out_c = outputs[0].size[1];
+        const int out_h = outputs[0].size[2];
+        const int out_w = outputs[0].size[3];
+        int out_count = outputs[0].total(1);
+        int inp_size = inputs.size();
+        if (have_reference) {
+            --inp_size;
+        }
+
+        float* out_data = outputs[0].ptr<float>();
+        int cntChannelsOfPrevious = 0;
+        for (int i = 0; i < inp_size; i++)
+        {
+            float* inp_data = inputs[i].ptr<float>();
+            int total = inputs[i].total();
+
+            const int num   = inputs[i].size[0];
+            const int inp_c = inputs[i].size[1];
+            const int inp_h = inputs[i].size[2];
+            const int inp_w = inputs[i].size[3];
+
+            float* top_data_chanoffsetptr = out_data + cntChannelsOfPrevious * out_h * out_w;
+            for (int n = 0; n < num; n++)
+            {
+                for (int c = 0; c < inp_c; c++)
+                {
+                    int cn = n * inp_c + c;
+                    for (int h = 0; h < out_h; h++)
+                    {
+                        for (int w = 0; w < out_w; w++)
+                        {
+                            float xpos = w / static_cast<float>(out_w - 1) * (inp_w - 1);
+                            float ypos = h / static_cast<float>(out_h - 1) * (inp_h - 1);
+
+                            // Get interpolated sample
+                            float tlx = floor(xpos);
+                            float tly = floor(ypos);
+                            float xdist = xpos - tlx;
+                            float ydist = ypos - tly;
+                            int src_idx = inp_w * (inp_h * cn + tly) + tlx;
+
+                            float sampleTL = inp_data[src_idx];
+                            float sampleTR = inp_data[min(src_idx + 1, total)];
+                            float sampleBL = inp_data[min(src_idx + inp_w, total)];
+                            float sampleBR = inp_data[min(src_idx + 1 + inp_w, total)];
+
+                            float sample = (1 - xdist) * (1 - ydist) * sampleTL +
+                                           (1 - xdist) * ydist       * sampleBL +
+                                            xdist      * (1 - ydist) * sampleTR +
+                                            xdist      * ydist       * sampleBR;
+
+                            int dst_idx = w + (out_w * h) + (out_w * out_h * c) + out_count * n;
+                            top_data_chanoffsetptr[dst_idx] = sample;
```

Comment:
Implementation looks very similar to one we have in resize_layer. Can we inherit this class from there?